### PR TITLE
feat: [CDS-56774]: BE support for selective stage execution in triggers

### PR DESCRIPTION
**Pull request by vcalasansh**
## Describe your changes
This PR adds support for selective stage execution in NG Triggers.

The stages to execute should be passed in a list called `stagesToExecute` that will be present in the root level of the Trigger's YAML. If the list is empty, all stages will be executed.

Sample Trigger YAML for selective stage execution:
```
trigger:
  name: triggerTestStages
  identifier: triggerTestStages
  enabled: true
  description: ""
  tags: {}
  orgIdentifier: default
  projectIdentifier: test
  pipelineIdentifier: Pipeline2Stages
  stagesToExecute:
    - stage1
    - stage2
  source:
    type: Webhook
    spec:
      type: Custom
      spec:
        payloadConditions: []
        headerConditions: []
  inputYaml: |
    pipeline:
      identifier: Pipeline2Stages
      stages:
        - stage:
            identifier: stage1
            type: Custom
            variables:
              - name: var1
                type: String
                value: abc
        - stage:
            identifier: stage2
            type: Custom
            variables:
              - name: var2
                type: String
                value: def
```

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/45389)

### CODE REVIEW

**Comment by vcalasansh** [comment]
trigger messagecheck

File: pipeline-service/service/src/main/java/io/harness/pms/triggers/TriggerExecutionHelper.java [hunk 1]
```diff
@@ -321,6 +323,27 @@ && isEmpty(triggerDetails.getNgTriggerConfigV2().getInputSetRefs())) {
               : templateMergeResponseDTO.getMergedPipelineYamlWithTemplateRef();
         }
 
+        StagesExecutionInfo stagesExecutionInfo = null;
```
Comment (sribalij) [inline pipeline-service/service/src/main/java/io/harness/pms/triggers/TriggerExecutionHelper.java:326]: Need to enable the flag?

File: pipeline-service/service/src/main/java/io/harness/pms/triggers/TriggerExecutionHelper.java [hunk 2]
```diff
@@ -399,8 +427,16 @@ private PlanExecution createPlanExecutionV2(TriggerDetails triggerDetails, Trigg
       try (PmsGitSyncBranchContextGuard ignore =
                pmsGitSyncHelper.createGitSyncBranchContextGuardFromBytes(gitSyncBranchContextByteString, false)) {
         RetryExecutionParameters retryExecutionParameters = RetryExecutionParameters.builder().isRetry(false).build();
-        ExecArgs execArgs = executionHelper.buildExecutionArgs(pipelineEntity, null, runtimeInputYaml,
-            Collections.emptyList(), Collections.emptyMap(), triggerInfo, null, retryExecutionParameters, false, false);
+        List<String> stagesToExecute = Collections.emptyList();
+        //        if (featureFlagService.isEnabled(
```
Comment (sribalij) [inline pipeline-service/service/src/main/java/io/harness/pms/triggers/TriggerExecutionHelper.java:431]: Need to enable the flag?

**Review by rathodmeetsatish — APPROVED** [review]
> Reviewed 6 of 11 files at r1, 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45389)* status: 11 file(s) to review. 1 of 2 approvals obtained.

**Review by sribalij — APPROVED** [review]
> Reviewed 6 of 11 files at r1, 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45389)* status: :shipit: complete! ✓

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 6 of 11 files at r1, 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45389)* status: :shipit: complete! ✓

**Comment by vcalasansh** [comment]
trigger review

**Pull request merged**